src/dc_receive_imf.rs

```diff
@@ -214,9 +225,15 @@ pub(crate) async fn dc_receive_imf_inner(
         contact::update_last_seen(context, from_id, sent_timestamp).await?;
     }
 
+    let received_msg = match received_msg {
+        Some(m) => m,
+        None => return Ok(None),
+    };
+
     // Update gossiped timestamp for the chat if someone else or our other device sent
     // Autocrypt-Gossip for all recipients in the chat to avoid sending Autocrypt-Gossip ourselves
     // and waste traffic.
+    let chat_id = received_msg.chat_id;
```

Comment:
Could be `received_msg.map_or(DC_CHAT_ID_TRASH, |received_msg| received_msg.chat_id)` if early exit above is removed.